### ISSUE TITLE
Please allow syncing between devices of Content-Type-Fixer data [feature request]

### ISSUE THREAD

**Issue by hub2git** [jscher2000/Content-Type-Fixer-extension#3]
Dear @jscher2000 ,

Thank you for a most useful extension.

Could I kindly ask you to implement [syncing ](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/storage/sync ) of Content-Type-Fixer data between devices?

This way, when I sign into my Firefox account at work or at a library, I don't have to customize my Content-Type-Fixer settings again.

My request was inspired after the Firefox Developers  [syncing for Multi-Account Containers.](https://blog.mozilla.org/security/2020/02/06/multi-account-containers-sync/) last week.

Thank you very much.

**Comment by jscher2000** [comment]
Thank you for the suggestion.

At some point, I definitely need to get a better understanding of how to properly use Sync storage. With local storage, I always know what is going on. With Sync, there will be discrepancies that I currently do not have any code to handle. 

For example, if a change arrives by Sync, in order for that to take effect, I need to re-read storage back into memory -- currently storage is only read when the extension initializes. But how will I know when to do that? 

Also, if you add or edit a content-type in your current session, when I save the change, I need to mesh all the data together and figure out what to do about inconsistencies. Should I ask the user? Et cetera.

**Comment by hub2git** [comment]
Hi @jscher2000,
Those are good questions. 

> For example, if a change arrives by Sync, in order for that to take effect, I need to re-read storage back into memory -- currently storage is only read when the extension initializes. But how will I know when to do that? But how will I know when to do that?

Hmmm... maybe have it check every X minutes? Or, isn't there a way for an extension to be alerted when there are changes in the Sync "domain"?

> Also, if you add or edit a content-type in your current session, when I save the change, I need to mesh all the data together and figure out what to do about inconsistencies. Should I ask the user? Et cetera.

Hmmm... that's a tough one. Yes, maybe a prompt telling the user that content-type A, B, and C have inconsistencies (and then the prompt will share the difference in layman's terms). Then the user can choose for each of the content-types:
[Option 1: Apply the values from Sync]
[Option 2: Keep the values that is on the local Firefox]
[Option 3: Merge(?!) (Dunno how that will look like, though)]
